pybleau/app/ui/dataframe_plot_manager_view.py

```diff
@@ -214,27 +213,38 @@ def _new_plot_button_fired(self):
         selector = PlotTypeSelector(
             view_klass=self.view_klass,
             num_containers=self.model.canvas_manager.num_container_managers,
-            container_idx=AUTO_TARGET_CONTAINER
+            container_idx=AUTO_TARGET_CONTAINER,
+            _plot_types=self.model.plot_types
         )
         ui = selector.edit_traits(kind="modal")
         if not ui.result:
             return
 
         plot_type = selector.plot_type
 
-        next_plot_num = len(self.model.contained_plots) + 1
-        if plot_type.startswith("Multi"):
-            new_plot_default_title = "Plot {i}"
+        if plot_type in list(self.model.custom_configs.keys()) and \
+                self.model.template_interactor is not None:
+            interactor = self.model.template_interactor
+            filepath = os.path.join(interactor.get_template_dir(), plot_type +
+                                    interactor.get_template_ext())
+            loader = interactor.get_template_loader()
+            configurator = loader(filepath)
```

Comment:
This would work, but `custom_configs` is a read-only property. This is done so that it is never out of sync with the files in the templates directory. Every call to `custom_configs` therefore gets the list anew from the directory; so users can add a file and then immediately make use of it, without us having to do any file system monitoring.
There may be an issue with storing the configurator, as well, since users are able to replace configurators with new ones that have the same name.

> I would store the configurator in the `self.model.custom_configs` so next time it's used, we don't need to load again. This could be done by checking the type of the value corresponding to plot_type and do the loading only if the value is a string.